### PR TITLE
Spoiler log

### PR DESCRIPTION
**Pull request by krnlyng**

### CODE REVIEW

**Comment by krnlyng** [comment]
I guess there's no interest in this?

**Comment by rrealmuto** [comment]
Sorry lol I had notifications off on this repo so never even saw this.

**Comment by SlyryD** [comment]
Would be cool to reference the ASM variables from C or vice versa to ensure they line up even if changes are made to the entrance list. Could you just declare `extern ... VAR_Entrances` in C?

**Comment by krnlyng** [comment]
> Would be cool to reference the ASM variables from C or vice versa to ensure they line up even if changes are made to the entrance list. Could you just declare `extern ... VAR_Entrances` in C?

I wasn't able to do so. I tried using extern in C but then the C code wasn't able to see the asm definition of VAR_Entrances, i tried the reverse also i.e. using extern in asm and making the variable global in C but that didn't work either.
I believe it's cause the rand.z64 file is included as a binary using incbin, but i'm not 100% sure what causes this.

**Comment by krnlyng** [comment]
I will make changes to only show the spoiler log when the filename contains a certain string once i find the time.

It was suggested that the spoiler log should only be included if the filename starts with, e.g. SL.
I wonder if the entrances should be seeded differently between a filename with SLX where X is something, and just X? I think that would make sense. I.e. should SL be part of the seed or not? I would say it should be.

**Comment by SlyryD** [comment]
Yes, I think SL should be part of the seed. That way when someone gets a race seed of 123456, they can't start up identical games at the same time with 123456 and SL123456.

**Comment by krnlyng** [comment]
Updated to only print the spoiler log if the file name starts with SL.

File: scripts/Patches.py [hunk 1]
```diff
@@ -427,4 +427,10 @@ def patch_rom(rom:Rom, options):
     save_context.write_save_table(rom)
 
     rom.write_int32(rom.sym('CFG_DUNGEON_INFO_ENABLE'), 1)
+
+    if options['spoiler']:
```
Comment (SlyryD) [inline scripts/Patches.py]: I'm okay with the spoiler being in the base rom now that we've established the "SL" convention. I don't think this needs to be a patching option.

**Comment by SlyryD** [comment]
Looks good to me! @rrealmuto

**Pull request merged**